src/js/oisc-warnings.js

```diff
@@ -0,0 +1,54 @@
+const initOISCWarnings = () => {
+  const supportsInterSectionObserver =
```

Comment:
And what happens in browsers that don't support it?

Reply:
> ⚠️ I've included my initial work on OISC warnings in here as a working example. Looking for reviews on the storybook structure at this stage, not this component. If we are happy with the approach I'll remove that work from this PR so we can merge this without shipping early work on the warnings component.